timesketch/lib/analyzers/win_crash.py

```diff
@@ -0,0 +1,195 @@
+        self.index_name = index_name
+        super(WinCrashSketchPlugin, self).__init__(index_name, sketch_id)
+
+    def formulate_query(self, elements):
+        """Generates the Elasticsearch query.
+
+        Args:
+            elements: Dictionary with a list of conditions
+
+        Returns:
+            The Elasticsearch query
+        """
+        conditions = list()
+        for e in elements.values():
+            conditions += ['({0})'.format(' AND '.join(e))]
+        return ' OR '.join(conditions)
+
+    def extract_filename(self, text):
+        """Finds filenames of crashed applications using a regular expression.
+
+        Args:
+            text: String that might contain the filename
+
+        Returns:
+            The string with filename if found
+        """
+        if '.exe' in str(text or '').lower():
+            match = self.FILENAME_REGEX.search(text)
+            if match:
+                # The regex can match on full file paths and filenames,
+                # so only return the filename.
+                return min([m for m in match.groups() if m])
+        return None
+
+    def mark_as_crash(self, event, filename):
+        """Mark entries with crash artefacts.
+
+        Args:
+            event: Elasticsearch event
+            filename: Application that crashed
+        """
+        if filename:
+            event.add_attributes({'crash_app': filename})
+        event.add_tags(['win_crash'])
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query = self.formulate_query(self.QUERY_ELEMENTS)
+
+        return_fields = ['data_type', 'message', 'filename']
+
+        # Generator of events based on your query.
+        events = self.event_stream(
+            query_string=query, return_fields=return_fields)
+
+        # Container for filenames of crashed applications.
+        filenames = list()
+
+        for event in events:
+            data_type = event.source.get('data_type')
+            event_text = None
+
+            # Search event log entries for filenames of crashed applications.
+            if data_type == 'windows:evtx:record':
+                event_text = event.source.get('message')
+
+            # Search file system entries for filenames of crashed applications.
+            elif data_type == 'fs:stat':
+                event_text = event.source.get('filename')
+
+            # Tag entries that show the crash reporting has been disabled.
+            elif data_type == 'windows:registry:key_value':
+                event.add_comment(
+                    'WARNING: The crash reporting was disabled. '
+                    'It could be indicative of attacker activity. '
+                    'For details refer to page 16 from '
+                    'https://assets.documentcloud.org/documents/3461560/'
+                    'Google-Aquarium-Clean.pdf.')
+                event.commit()
```

Comment:
in this case there is no need to continue evaluatingn the rest... I woudl rather opt for starting here

```
if data_type == 'windows:regist....`:
  event.add_comment(....
  event.commit()
  continue

if data_type == 'windows:evtx:record':
...

```
etc, etc, since now you are adding a comment, then doing a commit and then checkingn filename, which you know will not work, since you are just calling a function on an empty strign (or a none as the code stands now), so you might as well just do a `continue` so that you can just move ahead and evaluate the next record

Reply:
Changed the code as per your suggestion. I tried to avoid "continue" as it makes the code less readable for me.